predicators/envs/spot_env.py

```diff
@@ -2027,22 +2046,63 @@ class SpotBallAndCupStickyTableEnv(SpotRearrangementEnv):
 
     def __init__(self, use_gui: bool = True) -> None:
         super().__init__(use_gui)
-
         op_to_name = {o.name: o for o in _create_operators()}
-        # NOTE: we do not yet have planning operators sufficient enough
-        # to make the planning graph fully-connected. These are
-        # forthcoming.
+        op_to_name.update(
+            {o.name: o
+             for o in self._create_environment_specific_operators()})
         op_names_to_keep = {
             "MoveToReachObject", "MoveToHandViewObject",
             "MoveToBodyViewObject", "PickObjectFromTop", "PlaceObjectOnTop",
-            "DropObjectInsideContainerOnTop"
+            "DropObjectInsideContainerOnTop", "PickCupToDumpBall"
         }
         self._strips_operators = {op_to_name[o] for o in op_names_to_keep}
 
     @classmethod
     def get_name(cls) -> str:
         return "spot_ball_and_cup_sticky_table_env"
 
+    @staticmethod
+    def _create_environment_specific_operators() -> Iterator[STRIPSOperator]:
```

Comment:
this is reasonable, but it's inconsistent with how we're doing environment-specific operators in other environments (see `op_names_to_keep` in other envs). I kinda favor the other way (the `op_names_to_keep` way) because it's possible that we'll want to use the PickCupToDumpBall operator in some other environment too.

Reply:
in fact, if you just move this operator definition to the other operator definitions and remove this new method, I think you won't need any other changes